lib/awful/hotkeys_popup/widget.lua

```diff
@@ -285,6 +311,52 @@ function widget.new(args)
     end
 
 
+    function widget_instance:_abbreviate_hotkeys(target)
+        -- This is a very crude method and only useful for keygroups.
+        -- It compares the current key value without modifiers, as it would be
+        -- printed (e.g 2/3/4/5/6) with the string that the entire
```

Comment:
what about the case with 2/3/6/7?

Reply:
or if they keys are not sorted

Reply:
It won't be abbreviated. It will remain 2/3/6/7.

This code is really supposed to complement `awful.key` bindings with `keygroup`, which naturally creates bindings with the same description. 2/3/6/7 can only happen if the user individually creates bindings for 2, 3, 6, 7 while using the same description... which I don't think that happens very often.

It won't work with 7/6/5/4/3/2, either. Anyways, if it doesn't find a match, it won't try abbreviating them. No information is lost.

Reply:
> 2/3/6/7 can only happen if the user individually creates bindings for 2, 3, 6, 7 while using the same description... which I don't think that happens very often.

but it's not handled here as far as i can see from the function (or guide me to the exact line where it is, if i am mistaken)

Reply:
> It won't work with 7/6/5/4/3/2, either.

also i can't see from the code where it skips them

Reply:
i see now, it skips those cases by matching together with slashes

but that will also skip a valid cases like 1/3/2

Reply:
Yes, that's correct. Cases like 1/3/2/4/5 will be skipped because they're not in the correct order. (1/3/2 would be skipped anyways because it's only three keys.)

This only happens if users create bindings for each key separately, instead of using a keygroup. To fix it... I'd need to operate on tables.

Reply:
>  To fix it... I'd need to operate on tables.

yeah, that's exactly the point :+1:

Reply:
Please check #3001.